python/cudf_polars/cudf_polars/experimental/rapidsmpf/shuffle.py

```diff
@@ -126,40 +124,45 @@ def __exit__(
         _release_shuffle_id(self.op_id)
         return False
 
-    def insert_chunk(self, table: plc.Table) -> None:
+    def insert_chunk(self, chunk: TableChunk, stream: Stream) -> None:
```

Comment:
Hopefully we can remove the `stream` parameter here and just use `chunk.stream` for any operations here.

AFAICT, we should be fine to perform the `insert_chunk` on `chunk.stream`. Reading https://github.com/rapidsai/rapidsmpf/blob/decd6ca4d756f54cb9b57a7c874785894c4ba6a4/cpp/include/rapidsmpf/shuffler/shuffler.hpp#L98-L100, inserts should just require that "inserted buffers are stream-ordered with respect to their own stream", which calling `py_partition_and_pack(chunk.table_view(), ..., stream=chunk.stream)` should meet.